src/modules/functionality.js

```diff
@@ -0,0 +1,73 @@
+import addObjToLocalStorage from './objectToLS.js';
+import { store, taskArray } from './store.js';
+
+const submit = document.querySelector('.fa-right-from-bracket');
+const tasksContainer = document.querySelector('.tasks-container');
+const taskName = document.querySelector('.new-task');
+
+const addTask = (e) => {
+  e.preventDefault();
+  if (e.target.classList.contains('fa-right-from-bracket')) {
+    if (!taskName.value) return;
+    addObjToLocalStorage(taskName.value);
+    taskName.value = '';
+  }
+};
+
+const deleteTask = (e) => {
+  if (e.target.classList.contains('delete-icon')) {
+    const btn = e.target;
+    const removableTask = btn.closest('li');
+    const taskIndex = removableTask.dataset.indexNumber;
+
+    // Remove element from array
+    taskArray.splice(taskIndex - 1, 1);
+    localStorage.setItem('taskInput', taskArray);
```

Comment:
[OPTIONAL] Kindly consider using Javascript's `map()` function to update the `index` key after deleting an item.

```
    let count = 1;
  
    const newTask = taskArray.map((task) => {
      return { ...task, index: count ++ }; 
    });

    localStorage.setItem('taskInput', newTasks);
```

- This method will iterate through all the task objects and modify only the `index` value for each object by incrementing the variable `count` which always starts at 1.